Tests/SwiftDocCTests/Utility/ValidatedURLTests.swift

```diff
@@ -120,6 +120,13 @@ class ValidatedURLTests: XCTestCase {
             XCTAssertEqual(validatedWithHeading.components.path, expectedPath)
             XCTAssertEqual(validatedWithHeading.components.fragment, "Heading-Name")
         }
+        
+        for linkText in [
+            "url://com.example.test/new/docc=Whats%20New&version=DocC&Title=[Update]"
```

Comment:
Do we have a test sample of a URL that isn't percent-encoded already? I would want to make sure that that is covered. (Plus a couple for short strings or other kinds of edge cases that this issue has uncovered.)